### PR TITLE
Introduce the BTFhub module

### PR DESCRIPTION
**Pull request by mrproliu**
Follow Blog: https://kinvolk.io/blog/2022/03/btfgen-one-step-closer-to-truly-portable-ebpf-programs/
I see it could help users running the eBPF program with CO-RE on the old kernel(without recompile kernel), It's pretty cool. So we could add it.

### CODE REVIEW

File: scripts/build/bash/btfgen.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        elif [[ "$file" =~ $BPF_SO_PATTERN ]]
+        then
+          echo "-o $1/$file"
+        fi
+    done
+}
+
+${TMPDIR}/btfhub/tools/btfgen.sh -a ${ARCH} $(each_all_bpf_so_file $FROM)
+mkdir -p ${OUTPUT}
+cp -r ${TMPDIR}/btfhub/custom-archive/* ${OUTPUT}
```
Comment (kezhenxu94) [inline scripts/build/bash/btfgen.sh:50]: You actually copy some of the files from repo https://github.com/aquasecurity/btfhub-archive into our binary, please remember to add these in our binary license.

Comment (mrproliu) [reply scripts/build/bash/btfgen.sh:50]: The files under `custom-archive` are generated by `btfgen.sh`(btfhub tools), you mean we need to add https://github.com/aquasecurity/btfhub to the binary license? The `btfhub-archive` are only some archive binary files.  Am I right?

Comment (kezhenxu94) [reply scripts/build/bash/btfgen.sh:50]: > The files under `custom-archive` are generated by `btfgen.sh`(btfhub tools), you mean we need to add https://github.com/aquasecurity/btfhub to the binary license? The `btfhub-archive` are only some archive binary files. Am I right?

Correct me if I'm wrong, `btfgen.sh` "generates" `custom-archive` by selecting some of the files from https://github.com/aquasecurity/btfhub-archive , right? If so we should add license https://github.com/aquasecurity/btfhub-archive to binary license

Comment (mrproliu) [reply scripts/build/bash/btfgen.sh:50]: Yes, I take a look at this project. There are no licenses declared. So we ignore it temprory?

Comment (wu-sheng) [reply scripts/build/bash/btfgen.sh:50]: If it doesn't have a license, we should not use it. Default, the owner has its 100% copyright, and doesn't grant to us and the foundation, even though everything is readable, doesn't mean open-source. 

It is better if you open an issue to ask the owner.

Comment (wu-sheng) [reply scripts/build/bash/btfgen.sh:50]: OK, it is not urgent, let's hold this.

**Comment by vsxen** [comment]
cool!

File: scripts/build/bash/btfgen.sh (hunk 1 above)
Comment (rafaeldtinoco) [reply scripts/build/bash/btfgen.sh:50]: Ok license file created in the btfhub-archive repository. Sorry for the delays. Let me know if there is anything else needed.

Comment (mrproliu) [reply scripts/build/bash/btfgen.sh:50]: Thanks!!!

Comment (mrproliu) [reply scripts/build/bash/btfgen.sh:50]: @kezhenxu94 @wu-sheng Hi, the license issue has been resolved, please recheck:)

**Review by kezhenxu94 — APPROVED** [review]

File: scripts/build/bash/btfgen.sh (hunk 1 above)
Comment (wu-sheng) [reply scripts/build/bash/btfgen.sh:50]: @rafaeldtinoco Thanks for fixing this so quickly, and licensed in APLv2.

**Pull request merged**